packages/apps/api/src/handlers/user.ts

```diff
@@ -18,15 +17,15 @@ export function validatePassword(auth0Client: Auth0Client) {
         message: 'A password must be provided and non-blank',
       });
     }
-    const auth0Get: APIResponse<Auth0Profile> = await auth0Client.getUserByUserId(
+    const auth0Get: APIResponse<Auth0User> = await auth0Client.getUserByUserId(
       userId
     );
     if (auth0Get.status !== ResponseStatus.Success) {
       throw clientResponseToHttpError(auth0Get);
     }
 
     await checkPassword(
-      auth0Get.result.email,
+      auth0Get.result.email!,
```

Comment:
Am I right in thinking the exclamation mark casts this to `undefined` if the result is `null`? If so: when would `auth0Get.result.email` be `null`?

Reply:
`!` asserts that something that is typed as `T | undefined` is actually a `T` - for obvious reasons, the auth0 user type always has an `email` defined (I guess it's typed as `string | undefined` because some IDPs don't use emails?) so we make this assertion here.

Yesterday I convinced myself that it was preferable to assert at the point of use rather than at the module boundary, but I've changed my mind now and will do it in `Auth0Client`...